test/GitcoinPassportDecoder.ts

```diff
@@ -1243,43 +1028,119 @@ describe("GitcoinPassportDecoder", function () {
+      it("should revert if an cachedScore has an expiration date and is expired", async function () {
+        cachedScore.expirationTime = now - 3600 * 24;
+        await expect(
+          gitcoinPassportDecoderInternal.checkExpiration(cachedScore)
+        ).to.be.revertedWithCustomError(
+          gitcoinPassportDecoderInternal,
+          "AttestationExpired"
+        );
       });
 
-      it("should return false if an cachedScore has an expiration date and is expired", async function () {
-        cachedScore.expirationTime = now + 3600 * 24; // Unset expiration time
-        const isExpired =
-          await gitcoinPassportDecoderInternal.isCachedScoreExpired(
-            cachedScore
-          );
-        expect(isExpired).to.equal(false);
+      it("should not revert if an cachedScore has an expiration date and is not expired", async function () {
+        cachedScore.expirationTime = now + 3600 * 24;
+        await expect(
+          gitcoinPassportDecoderInternal.checkExpiration(cachedScore)
+        ).not.to.be.revertedWithCustomError(
+          gitcoinPassportDecoderInternal,
+          "AttestationExpired"
+        );
       });
 
-      it("should return true if a score has NO expiration date and expiration is determined based on issuance date", async function () {
+      it("should revert if a score has NO expiration date and expiration is determined based on issuance date", async function () {
         cachedScore.expirationTime = 0; // Unset expiration time
         cachedScore.time = now - maxScoreAge - 3600; // Make the attestation older than max age
-        const isExpired =
-          await gitcoinPassportDecoderInternal.isCachedScoreExpired(
-            cachedScore
-          );
-        expect(isExpired).to.equal(true);
+        await expect(
+          gitcoinPassportDecoderInternal.checkExpiration(cachedScore)
+        ).to.be.revertedWithCustomError(
+          gitcoinPassportDecoderInternal,
+          "AttestationExpired"
+        );
       });
 
-      it("should return false if a score has NO expiration date and it is not expired based based on issuance date and max age", async function () {
+      it("should not revert if a score has NO expiration date and it is not expired based based on issuance date and max age", async function () {
         cachedScore.expirationTime = 0; // Unset expiration time
         cachedScore.time = now - maxScoreAge + 3600; // Make the attestation younger than max age
-        const isExpired =
-          await gitcoinPassportDecoderInternal.isCachedScoreExpired(
-            cachedScore
-          );
-        expect(isExpired).to.equal(false);
+        await expect(
+          gitcoinPassportDecoderInternal.checkExpiration(cachedScore)
+        ).not.to.be.revertedWithCustomError(
+          gitcoinPassportDecoderInternal,
+          "AttestationExpired"
+        );
       });
     });
   });
+
+  describe("getPassport with ScoreV2 format", function () {
```

Comment:
I might be a bit paranoic, but I would add tests for all methods in from iGitcoinPassportDecoder to this test suit:

  function getPassport(
    address user
  ) external returns (Credential[] memory);
  function getScore(address user) external view returns (uint256);
  function getScore(uint32 communityId, address user) external view returns (uint256);
  function isHuman(address user) external view returns (bool);